From 5adde9af2763aab835653e7febcb7bf7fb6d6e10 Mon Sep 17 00:00:00 2001
From: Hau Vo <hau.vo.ra@renesas.com>
Date: Wed, 3 Jun 2026 07:46:04 +0700
Subject: [PATCH] configs: r8a78000_ironhide_defconfig: Disable
 LOCALVERSION_AUTO

Signed-off-by: Hau Vo <hau.vo.ra@renesas.com>

Upstream-Status: Inappropriate [R-Car specific feature]
---
 configs/r8a78000_ironhide_defconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configs/r8a78000_ironhide_defconfig b/configs/r8a78000_ironhide_defconfig
index 180704e75af..ecf83f06a33 100644
--- a/configs/r8a78000_ironhide_defconfig
+++ b/configs/r8a78000_ironhide_defconfig
@@ -41,3 +41,4 @@ CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_CLK_FREQ=1066666667
 CONFIG_UFS=y
 CONFIG_UFS_RENESAS_GEN5=y
+# CONFIG_LOCALVERSION_AUTO is not set
-- 
2.34.1

