From 5f3206b346407acb989c5e859d8b4e6aa3ac1db7 Mon Sep 17 00:00:00 2001
From: Nam Nguyen <nam.nguyen.nx@bp.renesas.com>
Date: Tue, 12 May 2026 22:52:53 +0700
Subject: [PATCH] 0001-Adding-ravb_hw_info-to-adapt-kernel-v6

---
 mch_core_main.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/mch_core_main.c b/mch_core_main.c
index 03d575e..eaa29f6 100644
--- a/mch_core_main.c
+++ b/mch_core_main.c
@@ -1049,12 +1049,15 @@ int mch_regist_interrupt(struct mch_private *priv,
 {
 	struct net_device *ndev = priv->ndev;
 	struct ravb_private *net_priv = netdev_priv(ndev);
+  const struct ravb_hw_info *info = net_priv->info;
 	struct device *dev = priv->dev;
 	const char  *irq_name;
 	int irq;
 	int err;
 
-	if (WARN_ON(net_priv->chip_id != RCAR_GEN3))
+  pr_info("info=%px multi_irqs=%d\n", info, info ? info->multi_irqs : -1);
+  
+	if (WARN_ON(!info->multi_irqs))
 		return -EPERM;
 
 	/* regist interrupt of avtp capture */
-- 
2.34.1

