diff --git a/ravb_proc.c b/ravb_proc.c
index c7a5b36..5328fa0 100644
--- a/ravb_proc.c
+++ b/ravb_proc.c
@@ -750,7 +750,7 @@ static int stats_show_driver_userpages(struct seq_file *m, void *v)
 
 static int stats_proc_open(struct inode *inode, struct file *file)
 {
-	return single_open(file, PDE_DATA(inode), NULL);
+	return single_open(file, pde_data(inode), NULL);
 }
 
 static const struct proc_ops stats_proc_fops = {
diff --git a/ravb_streaming_main.c b/ravb_streaming_main.c
index 81d2c7c..12d4f45 100644
--- a/ravb_streaming_main.c
+++ b/ravb_streaming_main.c
@@ -825,6 +825,7 @@ static inline void unregister_cbs_param(enum eavb_streamclass class,
 static int separation_filter_initialise(struct net_device *ndev)
 {
 	struct ravb_private *priv = netdev_priv(ndev);
+	const struct ravb_hw_info *info = priv->info;
 	u32 csr_ops;
 
 	csr_ops = ravb_read(ndev, CSR) & CSR_OPS;
@@ -835,7 +836,7 @@ static int separation_filter_initialise(struct net_device *ndev)
 		ravb_write(ndev, 0xffffffff, SFM0);
 		ravb_write(ndev, 0xffffffff, SFM1);
 	} else if (csr_ops == CSR_OPS_OPERATION) {
-		if (priv->chip_id == RCAR_GEN2)
+		if (!info->multi_irqs)
 			return -EBUSY;
 
 		/* Separation Filter Offset(OPERATION MODE) */
@@ -944,13 +945,13 @@ static void stq_release(struct kobject *kobj)
 static struct kobj_type stq_ktype_rx = {
 	.sysfs_ops = &stq_sysfs_ops,
 	.release = stq_release,
-	.default_attrs = stq_default_attrs_rx,
+	.default_groups = stq_default_attrs_rx,
 };
 
 static struct kobj_type stq_ktype_tx = {
 	.sysfs_ops = &stq_sysfs_ops,
 	.release = stq_release,
-	.default_attrs = stq_default_attrs_tx,
+	.default_groups = stq_default_attrs_tx,
 };
 
 static struct stqueue_info *get_stq(struct hwqueue_info *hwq, int index)
@@ -2179,7 +2180,8 @@ static long ravb_map_page_compat(struct file *file, unsigned long parm)
 	struct eavb_dma_alloc __user *pdma;
 	void __user *dma_vaddr;
 
-	pdma = compat_alloc_user_space(sizeof(*pdma));
+	// pdma = compat_alloc_user_space(sizeof(*pdma));
+	pdma = kmalloc(sizeof(*pdma), GFP_USER);
 
 	ret = ravb_map_page(file, (unsigned long)(pdma));
 	if (ret)
@@ -2213,7 +2215,8 @@ static long ravb_unmap_page_compat(struct file *file, unsigned long parm)
 	if (copy_from_user(&dma32, (void __user *)parm, sizeof(dma32)))
 		return -EFAULT;
 
-	pdma = compat_alloc_user_space(sizeof(*pdma));
+	// pdma = compat_alloc_user_space(sizeof(*pdma));
+	pdma = kmalloc(sizeof(*pdma), GFP_USER);
 	if (!access_ok(pdma, sizeof(*pdma)))
 		return -EFAULT;
 	if (__put_user(dma32.dma_paddr, &pdma->dma_paddr) ||
diff --git a/ravb_streaming_sysfs.c b/ravb_streaming_sysfs.c
index 449d3cf..7058cc4 100644
--- a/ravb_streaming_sysfs.c
+++ b/ravb_streaming_sysfs.c
@@ -439,14 +439,38 @@ static STQ_ATTR_RO(qno);
 /* for Tx stream */
 static STQ_ATTR_RO(cbs_params);
 
-struct attribute *stq_default_attrs_rx[] = {
+// const struct attribute_group *stq_default_attrs_rx[] = {
+// 	&stq_index_attribute.attr,
+// 	&stq_state_attribute.attr,
+// 	&stq_qno_attribute.attr,
+// 	NULL,
+// };
+
+// const struct attribute_group *stq_default_attrs_tx[] = {
+// 	&stq_index_attribute.attr,
+// 	&stq_state_attribute.attr,
+// 	&stq_qno_attribute.attr,
+// 	&stq_cbs_params_attribute.attr,
+// 	NULL,
+// };
+
+struct attribute *stq_attrs_rx[] = {
 	&stq_index_attribute.attr,
 	&stq_state_attribute.attr,
 	&stq_qno_attribute.attr,
 	NULL,
 };
 
-struct attribute *stq_default_attrs_tx[] = {
+const struct attribute_group stq_group_rx = {
+	.attrs = stq_attrs_rx,
+};
+
+const struct attribute_group *stq_default_attrs_rx[] = {
+	&stq_group_rx,
+	NULL,
+};
+
+struct attribute *stq_attrs_tx[] = {
 	&stq_index_attribute.attr,
 	&stq_state_attribute.attr,
 	&stq_qno_attribute.attr,
@@ -454,6 +478,15 @@ struct attribute *stq_default_attrs_tx[] = {
 	NULL,
 };
 
+const struct attribute_group stq_group_tx = {
+	.attrs = stq_attrs_tx,
+};
+
+const struct attribute_group *stq_default_attrs_tx[] = {
+	&stq_group_tx,
+	NULL,
+};
+
 #define STQ_STATS_SHOW_U64(_name) \
 static ssize_t stq_stats_##_name##_show(struct stqueue_info *stq, \
 			   struct stq_attribute *attr, char *page) \
diff --git a/ravb_streaming_sysfs.h b/ravb_streaming_sysfs.h
index 5ea04ff..56f6efc 100644
--- a/ravb_streaming_sysfs.h
+++ b/ravb_streaming_sysfs.h
@@ -77,8 +77,8 @@ extern const struct attribute_group *hwq_sysfs_groups_tx[];
  * stream queue info sysfs operations
  */
 extern const struct sysfs_ops stq_sysfs_ops;
-extern struct attribute *stq_default_attrs_rx[];
-extern struct attribute *stq_default_attrs_tx[];
+extern const struct attribute_group *stq_default_attrs_rx[];
+extern const struct attribute_group  *stq_default_attrs_tx[];
 extern struct attribute_group stq_dev_stat_group;
 
 #endif	/* #ifndef __RAVB_STREAMING_SYSFS_H__ */
