Upstream-Status: Inappropriate [oe specific]

---

diff --git a/libweston/meson.build b/libweston/meson.build
index 1af628d..d52c088 100644
--- a/libweston/meson.build
+++ b/libweston/meson.build
@@ -87,7 +87,7 @@
 subdir('desktop')
 subdir('shell-utils')
 
-dep_gbm = dependency('gbm', required: false, version: '>= 21.1.1')
+dep_gbm = dependency('gbm', required: false)
 
 if get_option('backend-vnc')
 	dep_pam = dependency('pam', required: false)
